provisioning/miniprov/hare_mp/store.py

```diff
@@ -109,6 +113,13 @@ def get_storage_set_nodes(self) -> List[str]:
 
         return storage_nodes
 
+    def search_val(self, parent_key: str, search_key: str,
+                   search_val: str) -> List[str]:
+        """
+        Searches a given key value under the given parent key.
+        """
+        return self.conf.search(self.index, parent_key, search_key, search_val)
```

Comment:
It's a rather strange API call. Do we really want to use it in our code? Is it beneficial?

Reply:
I agree to your claims, we have reported this but this what they have presently. We surely will report bugs  as we find them.